request.go

```diff
@@ -28,20 +28,47 @@ type Response struct {
 	Error  string `json:"error"`
 }
 
-// uriForAPI is to be called with something like "/v1/events" and it will give
-// the proper request URI to be posted to.
+// uriForAPI is to be called with either an API resource like "/v1/events"
+// or a full URL like the IP Ranges one
+// and it will give the proper request URI to be posted to.
 func (client *Client) uriForAPI(api string) (string, error) {
-	apiBase, err := url.Parse(client.baseUrl + "/api" + api)
+	var err error
+	// If api is a URI such as /v1/hosts/, /v2/dashboards... add credentials and return a properly formatted URL
+	if !strings.HasPrefix(api, "https://") || !strings.HasPrefix(api, "http://") {
```

Comment:
So I didn't actually try this out, but I don't think this condition is right. If the `api` value is e.g. `https://something`, the second part would still evaluate as `true` and so the `if` clause would actually be entered. I think what you need to do is `!(strings.HasPrefix(api, "https://") || strings.HasPrefix(api, "http://"))`